src/logging.ts

```diff
@@ -47,6 +47,24 @@ export function getLogLevel() {
   return logLevel;
 }
 
-export function setLogLevel(level: LogLevel) {
-  logLevel = level;
+function stringToLogLevel(level: string): LogLevel {
+  switch (level.toUpperCase()) {
+    case 'ERROR':
+      return LogLevel.ERROR;
+    case 'WARN':
+      return LogLevel.WARN;
+    case 'INFO':
+      return LogLevel.INFO;
+    case 'DEBUG':
+      return LogLevel.DEBUG;
+    case 'TRACE':
+      return LogLevel.TRACE;
+    default:
+      log(LogLevel.WARN, `Invalid log level: ${level}`);
+      return DEFAULT_LOG_LEVEL;
+  }
+}
+
+export function setLogLevel(level: string) {
+  logLevel = stringToLogLevel(level);
```

Comment:
NP: Simplification idea, feel free to ignore

```suggestion
export function setLogLevel(level: string) {
  if (!Object.values(LogLevel).includes(level.toUpperCase() as LogLevel)) {
    throw new Error(`Invalid log level: ${level}`);
  }
  logLevel = level.toUpperCase() as LogLevel;
```

Reply:
the only thing I don't like about this is the typecast, but it is much simpler. we could try and use a type guard instead? for now I'll just go ahead with what I have, but I am open in the real deal to do something better